### ISSUE TITLE
Incorrect authentication failure message handling

### ISSUE THREAD

**Issue by Perf** [lexik/LexikJWTAuthenticationBundle#448]
Hi,
when auth failure happens, at some point this code will be executed (in file `lexik/jwt-authentication-bundle/Security/Http/Authentication/AuthenticationFailureHandler.php`):
```php
    public function onAuthenticationFailure(Request $request, AuthenticationException $exception)
    {
        $event = new AuthenticationFailureEvent($exception, new JWTAuthenticationFailureResponse());

        $this->dispatcher->dispatch(Events::AUTHENTICATION_FAILURE, $event);

        return $event->getResponse();
    }
```
the issue is that `$exception` contains a specific error message, and `new JWTAuthenticationFailureResponse()` has the default message string 'Bad credentials'.

When event dispatched, response object is not updated and when after `return $event->getResponse();` is executed it returns unchanged response object that was created in the beginning of a method (`$event = new AuthenticationFailureEvent($exception, new JWTAuthenticationFailureResponse());`).

So, we have a specific error message, but showing to the user a default one.

**Comment by Perf** [comment]
As a quick fix - create your event subscriber.
```php
<?php

namespace App\EventSubscriber;

use Lexik\Bundle\JWTAuthenticationBundle\Event\AuthenticationFailureEvent;
use Lexik\Bundle\JWTAuthenticationBundle\Events;
use Lexik\Bundle\JWTAuthenticationBundle\Response\JWTAuthenticationFailureResponse;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;

class AuthenticationFailureSubscriber implements EventSubscriberInterface
{
    public static function getSubscribedEvents()
    {
        return [
            Events::AUTHENTICATION_FAILURE => 'onAuthenticationFailureResponse',
        ];
    }

    public function onAuthenticationFailureResponse(AuthenticationFailureEvent $event)
    {
        $message = $event->getException()->getMessage();
        if (empty($message)) {
            return;
        }
        /** @var JWTAuthenticationFailureResponse $response */
        $response = $event->getResponse();
        $response->setMessage($message);
        $event->setResponse($response);
    }
}
```

**Comment by chalasr** [comment]
Fixed by #472
